docs/source/usage.rst

```diff
@@ -0,0 +1,185 @@
+        --html-viewsource-base https://github.com/(user)/(repo)/tree/(tag_or_commit_sha) \
+        --make-html \
+        --html-output=apidocs \
+        --project-base-dir "$(pwd)" \
+        --docformat=restructuredtext 
+
+.. note:: This exemple assume you are running the ``pydoctor`` build from Unix and the current directory is the root folder of your package. 
+
+.. warning:: The ``--html-viewsource-base`` argument  should point to a tag or a commit SHA rather than a branch since line 
+    numbers aren't going to match otherwise when commits are added to the branch after the documentation has been published.
+
+Publish your documentation with Github actions
+----------------------------------------------
+
+It is the best practice to automatically generate and publish your documentation through your favorite CI. 
+
+Here is an exemple of how to do it with Github actions and publish the documentation to the default Github pages website, using the third party action ``peaceiris/actions-gh-pages`` (`documentation`__).
+
+__ https://github.com/peaceiris/actions-gh-pages
+
+::
+
+    name: publish-pydoctor-apidocs
+    on:
+    - push
+
+    jobs:
+      deploy:
+        runs-on: ubuntu-latest
+
+        steps:
+        - uses: actions/checkout@master
+        - name: Set up Python 3.8
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.8
+        
+        - name: Install package
+          run: |
+            python -m pip install --upgrade pip setuptools wheel
+            python -m pip install .
+            python -m pip install pydoctor
+
+        - name: Generate pydoctor documentation
+          run: |
+            # Allow pydoctor to exit with non-zero status code
+            set +e
+
+            # Run pydoctor build
+            pydoctor \
+                --add-package=(packagedirectory) \
+                --project-name=(projectname) \
+                --project-url=https://github.com/$GITHUB_REPOSITORY \
+                --html-viewsource-base https://github.com/$GITHUB_REPOSITORY/tree/$GITHUB_SHA \
+                --make-html \
+                --html-output=./apidocs \
+                --project-base-dir "$(pwd)" \
+                --docformat=restructuredtext 
+
+        - name: Publish pydoctor documentation to the gh-pages branch
+          uses: peaceiris/actions-gh-pages@v3
+          with:
+            github_token: ${{ secrets.GITHUB_TOKEN }}
+            publish_dir: ./apidocs
+            commit_message: "Generate pydoctor documentation"
+
+.. note:: As mentionned in the ``actions-gh-pages`` documentation, the first workflow run won't actually publish the documentation to Github pages. 
+    Github pages needs to be enabled afterwards in the repo settings, select ``gh-pages`` branch, then re-run your workflow. 
+
+    The website we'll be at https://(user).github.io/(repo)/
+
+Publish your documentation with Sphinx and Read The Docs
```

Comment:
maybe remove this section for now...and add it once we create the Read The Docs Sphinx extension